drivers/lora/sx1276.c

```diff
@@ -0,0 +1,492 @@
+/*
+ * Copyright (c) 2019 Manivannan Sadhasivam
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/counter.h>
+#include <drivers/gpio.h>
+#include <drivers/lora.h>
+#include <spi.h>
+#include <zephyr.h>
+
+#include <sx1276/sx1276.h>
+
+#define LOG_LEVEL CONFIG_LORA_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(sx1276);
+
+#define SX1276_MAX_DIO		5
+
+#define GPIO_RESET_PIN		DT_INST_0_SEMTECH_SX1276_RESET_GPIOS_PIN
+#define GPIO_CS_PIN		DT_INST_0_SEMTECH_SX1276_CS_GPIOS_PIN
+
+#define SX1276_REG_PA_CONFIG			0x09
+#define SX1276_REG_PA_DAC			0x4d
+#define SX1276_REG_VERSION			0x42
+
+extern DioIrqHandler *DioIrq[];
+
+int sx1276_dio_pins[SX1276_MAX_DIO] = {
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_PIN_0,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_PIN_1,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_PIN_2,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_PIN_3,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_PIN_4,
+};
+
+static char sx1276_dio_ports[SX1276_MAX_DIO][6] = {
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_CONTROLLER_0,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_CONTROLLER_1,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_CONTROLLER_2,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_CONTROLLER_3,
+	DT_INST_0_SEMTECH_SX1276_DIO_GPIOS_CONTROLLER_4,
+};
+
+struct sx1276_data {
+	struct device *counter;
+	struct device *reset;
+	struct device *spi;
+	struct spi_config spi_cfg;
+	struct device *dio_dev[SX1276_MAX_DIO];
+	struct k_sem data_sem;
+	RadioEvents_t sx1276_event;
+	u8_t *rx_buf;
+	u8_t rx_len;
+} dev_data;
+
+bool SX1276CheckRfFrequency(uint32_t frequency)
+{
+	/* TODO */
+	return true;
+}
+
+void RtcStopAlarm(void)
+{
+	counter_stop(dev_data.counter);
+}
+
+void SX1276SetAntSwLowPower(bool status)
+{
+	/* TODO */
+}
+
+void SX1276SetBoardTcxo(u8_t state)
+{
+	/* TODO */
+}
+
+void SX1276SetAntSw(u8_t opMode)
+{
+	/* TODO */
+}
+
+void SX1276Reset(void)
+{
+	gpio_pin_configure(dev_data.reset, GPIO_RESET_PIN,
+			   GPIO_DIR_OUT | GPIO_PUD_NORMAL);
+	gpio_pin_write(dev_data.reset, GPIO_RESET_PIN, 0);
+
+	k_sleep(1);
+
+	gpio_pin_configure(dev_data.reset, GPIO_RESET_PIN,
+			   GPIO_DIR_IN | GPIO_PUD_NORMAL);
+	k_sleep(6);
+}
+
+void BoardCriticalSectionBegin(uint32_t *mask)
+{
+	*mask = irq_lock();
+}
+
+void BoardCriticalSectionEnd(uint32_t *mask)
+{
+	irq_unlock(*mask);
+}
+
+uint32_t RtcGetTimerElapsedTime(void)
+{
+	return counter_read(dev_data.counter);
+}
+
+u32_t RtcGetMinimumTimeout(void)
+{
+	/* TODO: Get this value from counter driver */
+	return 3;
+}
+
+void RtcSetAlarm(uint32_t timeout)
```

Comment:
From rtc-board.h:
```C
 /*!
 * \brief Sets the alarm
 *
 * \note The alarm is set at now (read in this funtion) + timeout
 *
 * \param timeout [IN] Duration of the Timer ticks
 */
void RtcSetAlarm( uint32_t timeout );
```
I think you are mixint up ticks and us here.

Reply:
Ah, yes indeed! Will fix it.